packages/components/src/navigation/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Animate from '../animate';
+import { NavigationContext } from './context';
+import { DEFAULT_LEVEL } from './constants';
+import { Root } from './styles/navigation-styles';
+
+export default function Navigation( {
+	activeItem,
+	activeLevel = DEFAULT_LEVEL,
+	children,
+	className,
+	setActiveItem = noop,
+	setActiveLevel = noop,
+} ) {
+	const [ item, setItem ] = useState( activeItem );
+	const [ level, setLevel ] = useState( activeLevel );
+	const [ slideOrigin, setSlideOrigin ] = useState();
+
+	const activateItem = ( itemId ) => {
+		setItem( itemId );
+		setActiveItem( itemId );
+	};
+
+	const activateLevel = ( levelId, slideInOrigin = 'left' ) => {
+		setSlideOrigin( slideInOrigin );
+		setLevel( levelId );
+		setActiveLevel( levelId );
+	};
+
+	const isMounted = useRef( false );
```

Comment:
I think that comment explaining why this is needed would be helpful (prevent animation on mount).